backend/git_connectors/views/gitlab.py

```diff
@@ -187,35 +187,37 @@ def get(self, request: Request, id: str):
             git_app = (
                 GitApp.objects.filter(gitlab__id=id).select_related("gitlab").get()
             )
+        
+            gl_app = cast(GitlabApp, git_app.gitlab)
+            access_token = GitlabApp.ensure_fresh_access_token(gl_app)
+            url = f"{gl_app.gitlab_url}/api/v4/projects"
+            params = {
+                "membership": "true",
+            }
+            headers = {
+                "Accept": "application/json",
+                "Authorization": f"Bearer {access_token}",
+            }
+            response = requests.get(
+                url + "?" + urlencode(params, doseq=True), headers=headers
+            )
+
+            response.raise_for_status()
+
+            return Response(
+                data={
+                    # `x-total`` will not show if there is more than 10000+ repos,
+                    #  so we send just 10 001 to signal it
+                    "repositories_count": int(response.headers.get("x-total", 10_001)),
+                }
+            )
         except GitApp.DoesNotExist:
             raise exceptions.NotFound(f"Gitlab app with id {id} does not exist")
-
-        gl_app = cast(GitlabApp, git_app.gitlab)
-        access_token = GitlabApp.ensure_fresh_access_token(gl_app)
-        url = f"{gl_app.gitlab_url}/api/v4/projects"
-        params = {
-            "membership": "true",
-        }
-        headers = {
-            "Accept": "application/json",
-            "Authorization": f"Bearer {access_token}",
-        }
-        response = requests.get(
-            url + "?" + urlencode(params, doseq=True), headers=headers
-        )
-        if not status.is_success(response.status_code):
+        except Exception:
```

Comment:
```suggestion
        except HTTPError as error:
        	print(
                error.message,
                error.response.status_code,
                error.response.text,
             )
```

It might be better to catch the errors that we are sure means the app may not be installed correctly instead of using a catch all here. If there is another legit error this will silence it.

Also, we can log this error in case this happens in the future.